format.go

```diff
@@ -99,8 +100,8 @@ func (v *Value) normalize() bool {
 	switch v2 := v.Value.(type) {
 	case Literal:
 		// Format string.
-		if v2.Kind() == '"' && bytes.IndexByte(v2, '\\') >= 0 {
```

Comment:
I have _no_ context for why we were only doing string normalization if the string included a backslash, so this may have unintended consequences I don't know about. Plz advise

Reply:
The `bytes.IndexByte(v2, '\\') >= 0` expression is an optimization that avoids normalizing the string if there are no escape characters present. I should have commented on that.

`v.Value = String(strings.TrimSpace(v2.String()))` is changing the semantic meaning of a JSON string. It makes sense in the context of how Tailscale uses HuJSON in ACLs, but doesn't seem to belong within the general purpose HuJSON formatter where surrounding whitespace may be sensible for some particular use-case. We wouldn't want to change the semantic meaning for that.

An alternative to this is to expose a `Value.Range` method that iterates through all values in a depth-first order. Thus, our control server can do something like:
```go
// Strip surrounding whitespace from all JSON strings.
// These never matter in the context of our ACL file.
v.Range(func(v *Value) bool {
    if v2, ok := v.Value.(Literal); ok && v2.Kind() == '"' {
        v2.Value = String(strings.TrimSpace(v2.String()))
    }
    return true
})
v.Format()
```

Even then, I'm not sure how this helps tailscale/tailscale#6040. #6040 could manifest itself without any whitespace. If user validation rejected any users with commas, then that would be sufficient. I hope comma is not a valid character ever.

Reply:
Aha! That's better, will change + add into comments. There's a separate / tighter check for additional invalid characters in corp, but in cases of commas or mid-string spaces we'd want to warn the users to have them fix it themselves, whereas leading/trailing spaces seemed safe to discard. Will close and open a new one with the suggested fix instead, ty!